### ISSUE TITLE
CorsixTH crash with updated Sound-1.dat french file

### ISSUE THREAD

**Issue by legluondunet** [CorsixTH/CorsixTH#665]
Hello,

in this unofficial patch:
http://hospital.ragondin-earth.org/
you have a sound file: sound-1.dat wich corrects several french sound bugs:
Google translation:
"Using a utility that allows you to read / change the sounds of Theme Hospital, I found, to my surprise, that new sounds were present in the English version of the game that we could not find in other languages of the game. in addition, I found that some ads from the receptionist was slightly incorrect, especially about the message saying that a medical researcher had expected an electrolysis chamber while a regular doctor could suffice, or , the service call to the arrival of patients victims of "heart-breaking." I also found a strange bug in relation to incorrect locations in the sound file on some calls from the receptionist, especially about excuses or service call messages. Unfortunately, in this case, I found that only 5 service calls work, the others being silent for some reason."

But if I install this file Corsix crash at start.
## This window will display useful information if an error occurs.

Erreur de segmentation (core dumped)

It will be greatly appreciate by french TH gamers if CorsixTH could could tolerate this file.

LGDN.

**Comment by SimplyRyan** [comment]
I can't speak for the file in question but if you could play CorsixTH without this modification and report any issues then we can make the required fixes (Its also worth noting that the original game had this bug and CorsixTH simply doesn't replicate the issue.). We already correct a few text issues with the orignal game by using overrides in the French lanauge file, you can see these here - https://github.com/CorsixTH/CorsixTH/blob/master/CorsixTH/Lua/languages/french.lua

**Comment by legluondunet** [comment]
This patch seems heavily used by french users. I opened this issue for several objects:
- let a trace on the internet of incompatibility between this patch and Corsixth
- inform Corsixth team 
A crash give only one information to the user: that software doesn't work. Perrhaps you could add to corsixth a file verification that prevent users to use modified files? Or inform them, at the corsixth start, that they are using a patch that is not supported by corsixth and could crash it.

**Comment by MarkL1961** [comment]
If you are having problems with sound effects, there was an issue that became known after 0.40 was released as one or more sound effects relating to the gates of hell were causing a crash.
If that is what you are having a problem with, then this was fixed by this [update](https://github.com/CorsixTH/CorsixTH/pull/600) in the trunk.

**Comment by legluondunet** [comment]
MarkL1961, I'm using the last CorsixTH git, so your update does not work with this sound file. Perhaps you could test it?
http://hospital.ragondin-earth.org/LANG-1.DAT

**Comment by MarkL1961** [comment]
You should remove that link please as it could be taken as distributing the original files.
Lang is the written text, such as what the advisor tells you and menu names etc.
It should not cause the game to crash, but there have been issues with the correctness of some translations.  As Ryan as already said we can override those sorts of mistakes and already have done when they have been pointed out to us.
If it sound effects or announcements etc then it is sound-1.dat as in the title of this issue that needs to be checked

**Comment by MarkL1961** [comment]
I am not getting any crash from the language file.  I can't comment on the actual text as I can't read or speak French.

**Comment by legluondunet** [comment]
you tested it with the last corsixTH git?

**Comment by MarkL1961** [comment]
No, against 0.40.  Do you want me to test against trunk as well?

**Comment by MarkL1961** [comment]
Tested with http://www.armedpineapple.co.uk/cth/builds/CTH-20150210-master-cbfedc04d8.zip and all seems fine

**Comment by legluondunet** [comment]
ok thank you all :).

**Comment by MarkL1961** [comment]
Does this mean your problem was not with the language or sound file, but the missing .dll's?

**Comment by legluondunet** [comment]
@MarkL1961 Mark I just compiled the last git version, applied the patch and CorsixTH crash at start, without another message than "segmentation fault, core dumped". But If I replace the "sound-1.dat" by the original one, CorsixTH no more crash. I closed this issue because I don't want to annoying dev with bugs from an unofficial patch.

**Comment by MarkL1961** [comment]
What is this patch doing?  If it is to correct text, we can do that, if it is to correct sounds then you are best sticking with the original for now.  Only you have mentioned sound-1.dat and also LANG-1.DAT, which is the language file (which is what you linked to).

**Comment by legluondunet** [comment]
Mark I give a link to the wrong file, it's sound-1.dat from this patch wich make CTH crash, not LANG-1.DAT. In the original french version of the sound-1.dat file, some sounds from the receptionist are not correct or missing, this new sound data corrects that.

**Comment by MarkL1961** [comment]
I think for now you will have to put up with the missing sounds.  Perhaps one day we will have our own, but for now I am not aware of anyone working on this.

**Comment by lewri** [comment]
Do we still wish to pursue supporting this? I'm not sure what specific part of the patched dat file leads to crashing, but I would guess the `Level 5 fix` only version should work out the box?

**Comment by ARGAMX** [comment]
I double‑checked this task.

I downloaded both `LANG-1.DAT` and `SOUND-1.DAT` (`THPatch.rar`) via the provided link.
I installed them and launched `CorsixTH 0.69.2`.
Everything’s great — the whole game is in French, the texts are in French, the voice‑over is in French, I don’t understand a single word in French. But everything works — and that’s good.
No crashes occur.

Next, I tried `CorsixTH 0.70.1`.
The texts are in French, so `LANG-1.DAT` works.
But the sound is completely gone. 
That is, this French `SOUND-1.DAT` doesn’t work in `CorsixTH 0.70.1`.
This file lost compatibility starting from `CorsixTH 0.70.0`.
However, the game doesn’t crash.
Strange. 

But in any case, this is a different problem. 
No one has reported about it yet — it turns out I’m the first to report about it. 
But I believe this is already outside of the scope of this task. 
If someone needs support for this `SOUND-1.DAT` file, please create a new task.
